### PR TITLE
Added time interval times object

### LINKED ISSUE #25352: Add Times class for a time interval
## Reason
This would allow users to output at a certain time duration frequency, for example.

## Design
This would create a new Times class `TimeIntervalTimes`.

## Impact
New class, no impact to existing input.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #25352

### CODE REVIEW

File: framework/src/times/TimeIntervalTimes.C [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          "If the parameter 'end_time' is not provided, the executioner type must be 'Transient'.");
+  }
+
+  if (MooseUtils::absoluteFuzzyLessEqual(end_time, start_time))
+    mooseError("The end time must be greater than the start time.");
+
+  const auto time_interval = getParam<Real>("time_interval");
+  const bool always_include_end_time = getParam<bool>("always_include_end_time");
+
+  _times.push_back(start_time);
```
Comment (GiudGiud) [inline framework/src/times/TimeIntervalTimes.C:71]: need to clear first because people could be calling this twice (once per execution of the object)

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: framework/src/times/TimeIntervalTimes.C (hunk 1 above)
Comment (joshuahansel) [reply framework/src/times/TimeIntervalTimes.C:71]: I'm confused - everything is in the constructor, so how does it get called twice? I'm following some of the other Times that do everything in the constructor.

**Review by GiudGiud — DISMISSED** [review]

**Review by GiudGiud — APPROVED** [review]
love the battery of tests

**Comment by joshuahansel** [comment]
> love the battery of tests

This brings up a question I had: do we ever have "too many" tests, considering run time of the suite? In this case, the tests don't take long to run, but I realized I could have put multiple times in a single test.

**Pull request merged**